explorer/interpreter/interpreter.cpp

```diff
@@ -747,9 +770,12 @@ auto Interpreter::Convert(Nonnull<const Value*> value,
                                            &array_type.element_type());
           break;
         }
-        default:
-          CARBON_FATAL() << "Can't convert value " << *value << " to type "
-                         << *destination_type;
+        default: {
+          CARBON_CHECK(isa<VariableType, AssociatedConstant>(destination_type))
```

Comment:
As above: when you're already switching on the Kind, shouldn't we prefer cases?

Reply:
Let me know if you're unhappy with the new approach.